ntbs-service/Services/NotificationService.cs

```diff
@@ -86,41 +87,41 @@ public async Task UpdateClinicalDetailsAsync(Notification notification, Clinical
         {
             context.Entry(notification.ClinicalDetails).CurrentValues.SetValues(clinicalDetails);
 
-            await context.SaveChangesAsync();
+            await UpdateDatabase();
         }
 
         public async Task UpdateEpisodeAsync(Notification notification, Episode episode)
         {
             context.Entry(notification.Episode).CurrentValues.SetValues(episode);
 
-            await context.SaveChangesAsync();
+            await UpdateDatabase();
         }
 
         public async Task UpdateContactTracingAsync(Notification notification, ContactTracing contactTracing) {
             context.Entry(notification.ContactTracing).CurrentValues.SetValues(contactTracing);
 
-            await context.SaveChangesAsync();
+            await UpdateDatabase();
         }
 
         public async Task UpdatePatientTBHistoryAsync(Notification notification, PatientTBHistory tBHistory)
         {
             context.Entry(notification.PatientTBHistory).CurrentValues.SetValues(tBHistory);
 
-            await context.SaveChangesAsync();
+            await UpdateDatabase();
         }
 
         public async Task UpdateSocialRiskFactorsAsync(Notification notification, SocialRiskFactors socialRiskFactors)
         {
             UpdateSocialRiskFactorsFlags(socialRiskFactors);
```

Comment:
Turns out that this could actually be simplified :)

Reply:
Nice!